### PR TITLE
[mithril]: add responseType to RequestOptions

### PR DESCRIPTION
**Pull request by aecay**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://mithril.js.org/request.html#signature
- [x] <s>If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.</s> Not applicable
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] <s>If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.</s> Not applicable

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## mithril ([<kbd>unpkg</kbd>](https://unpkg.com/browse/mithril@latest/))
was missing the following properties:
1. m
1. vnode
1. PromisePolyfill
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ddd31079c78ca080c2370b1bc1f69dcedd7cce1b
</p>

File: types/mithril/index.d.ts [hunk 1]
```diff
@@ -138,6 +138,8 @@ declare namespace Mithril {
         background?: boolean;
         /** Milliseconds a request can take before automatically being terminated. */
         timeout?: number;
+        /** The expected type of the response, as a legal value of XMLHttpRequest.responseType. */
+        responseType?: "" | "arraybuffer" | "blob" | "document" | "json" | "text";
```
Comment (isiahmeadows) [inline types/mithril/index.d.ts:142]: Let's reuse the existing DOM type - we support everything the browser does.

```suggestion
        responseType?: XMLHttpRequestResponseType;
```

**Review by dead-claudia — CHANGES_REQUESTED** [review]

**Comment by isiahmeadows** [comment]
@ DTS team: is there a way to tell that Danger checker that its warnings are illegitimate?

- `m` is exposed in our global type defs but not `index.d.ts` as it's not assigned when importing.
- The rest are false positives - they aren't exported from our bundle.

File: types/mithril/index.d.ts (hunk 1 above)
Comment (aecay) [reply types/mithril/index.d.ts]: Thanks for the suggestion 🙂 It's much better your way.

**Comment by aecay** [comment]
@isiahmeadows the CI run after committing your fix led to
```
Error: File ./types/markedjs__html-differ/lib/logger.d.ts listed in OTHER_FILES.txt is already reachable from tsconfig.json.
```

As far as I can see, this is a file in a completely different library's typings, and I can't see any reason for the change to affect it.  But, I don't really know much about the CI jobs in this repo.  Do you have any idea of what might be going on? 😕

**Comment by isiahmeadows** [comment]
@aecay I don't run this repo, so not sure.

**Review by dead-claudia — APPROVED** [review]
LGTM, test failures unrelated

**Comment by isiahmeadows** [comment]
@aecay Rebase against current `master` and push again. The DTS people made some recent changes and that might resolve the errors.

**Review by dead-claudia — APPROVED** [review]

**Comment by isiahmeadows** [comment]
Ready to merge

**Pull request merged**